### PR TITLE
Added delete item button on list componenet

### LINKED ISSUE #11: 11. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future.
It’s possible our users may make an error when entering an item or may decide they won’t be buying a certain item again in the future. Let’s help them keep their lists tidy!

AC:

* User is able to delete an item from the shopping list
* Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
* Deletion should cause the associated record(s) in the database to be deleted

### PR DESCRIPTION
**Pull request by MabelM1**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

We updated the List.js component to include a button for each item. When the button is clicked, we call a new deleteItem function. This function shows a confirm prompt on the window. If the user confirms that they want to delete the item, we delete the item from the database. 

## Related Issue

closes #11 

## Acceptance Criteria

- User is able to delete an item from the shopping list

- Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions

- Deletion should cause the associated record(s) in the database to be deleted

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Screen Shot 2020-08-26 at 7 48 47 PM

### After

Screen Shot 2020-09-09 at 6 50 00 PM

## Testing Steps / QA Criteria

Press the delete button - a window confirm modal should appear. Pressing ok should remove the item, pressing cancel should not remove the item.

### CODE REVIEW

File: src/List.js [hunk 1]
```diff
@@ -47,6 +47,25 @@ const List = () => {
...
+        .delete()
+        .then(function() {
+          console.log('Document successfully deleted!');
+        })
+        .catch(function(error) {
+          console.error('Error removing document: ', error);
+        });
+    }
+  };
+
```
Comment (lacrivella) [inline src/List.js:68]: love it! nice and clean code.

**Review by lacrivella — APPROVED** [review]
Looks great! Nice clean code.

File: src/List.js (hunk 1 above)
Comment (lacrivella) [inline src/List.js:68]: y'all make it look so easy! I like how you only hit the database after the user confirms on the alert window.

**Review by stacietaylorcima — APPROVED** [review]
Amazing, crystal clear work here, friends! Left some comments with my approval. ✅

File: src/List.js [hunk 2]
```diff
@@ -47,6 +47,25 @@ const List = () => {
     setfilteredItems(filterItems);
   };
 
+  const deleteItem = e => {
+    const deleteConfirm = window.confirm(
+      'Are you sure you want to delete this item?',
+    );
+    if (deleteConfirm === true) {
```
[the remark this answers is not in the document]
Comment (MabelM1) [reply src/List.js]: Thank you! I updated it!

**Pull request merged**